airsonic-main/src/main/java/org/airsonic/player/controller/MusicFolderSettingsController.java

```diff
@@ -105,14 +105,14 @@ private void expunge() {
 
         // to be before dao#expunge
         MediaLibraryStatistics statistics = indexManager.getStatistics();
-        if (statistics != null) {
+        if (statistics != null && !mediaScannerService.isScanning()) {
```

Comment:
Executes exclusive control in more large block.
You can separate separate the statuses for log, but that's enough.